pipeline/backend/kubernetes/pod.go

```diff
@@ -43,7 +43,7 @@ func Pod(namespace string, step *types.Step, labels, annotations map[string]stri
 		}
 	}
 
-	pullPolicy := v1.PullIfNotPresent
+	var pullPolicy v1.PullPolicy
```

Comment:
Pro: the policy can be set by cluster
Con: if policy is by default no-pull and image dont existvlocally step will fail

Reply:
> if policy is by default no-pull and image dont exist locally step will fail

Do you mean `Never` by `no-pull`?

I didn't find how default pull policy can be changed. If you know, please, refer to it.

There may be some admission controller like [AlwaysPullImages](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#alwayspullimages), but for `Never`. In this case administrator explicitly sets this and know what he's doing. He have to populate cache with desired image. As in general, not only with woodpecker.

Reply:
Maybe there Kuberenetes implementations exist, which doesn't follow official docs...

Anyway, suppose cluster have different default image pull policy. In this case we can introduce configuration parameter `WOODPECKER_BACKEND_K8S_DEFAULT_POD_IMAGE_PULL_POLICY`.

Then if step's `pull` is not defined, look into `WOODPECKER_BACKEND_K8S_DEFAULT_POD_IMAGE_PULL_POLICY`. If `WOODPECKER_BACKEND_K8S_DEFAULT_POD_IMAGE_PULL_POLICY` is not defined, then `pullPolicy == nil` and using cluster default.